Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/survey/Surveys.kt

```diff
@@ -70,12 +71,13 @@ class Surveys @Inject constructor(
     suspend fun requestDetails(type: Type): Survey {
         val config = appConfigProvider.getAppConfig().survey
         Timber.v("Requested survey: %s", config)
-        val now = timeStamper.nowUTC
+        val now = timeStamper.nowJavaUTC
 
         oneTimePasswordRepo.otpAuthorizationResult?.apply {
-            if (authorized && redeemedAt.toDateTime().monthOfYear() == now.toDateTime().monthOfYear()) {
+            if (authorized && redeemedAt.toLocalDateUtc().month == now.toLocalDateUtc().month) {
```

Comment:
Before change DateTime was used, not LocalDate, I think `ZonedDateTime ` should be used here. it is true that month is referenced here , but if the time is at end of month I think it would be problematic , wouldn’t be ?

Reply:
@chiljamgossow  can you please check that?

Reply:
what about ` YearMonth.from(redeemedAt) == YearMonth.from(now)` ?

Reply:
The logic before was already flawed, as Mohamed mentioned there is an issue with comparing only the month but not the year. 
Juraj is right too, toLocalDateUtc() is actually better than it was before, as device time zone can change and we use utc in many places for that reason. So add a year check as well? (could also fix it separately as is a bug that was there before)

Reply:
I've added a year check